### ISSUE TITLE
[CI] Provide measurements for IGC 2.0 transition

### ISSUE THREAD

**Issue by esukhov** [intel/intel-xpu-backend-for-triton#2125]
For IGC2.0 transition, we want to estimate the performance impact of removing of the recompilation path.
The transition will have some performance impact, so we are trying to estimate and prepare. 

Could you please make a measurements on Triton benchmarks for this 2 versions of driver:

open-linux-driver-verify-dev_igc-86345 --> recompilation disabled
open-linux-driver-ci-dev_igc-17627 --> reference

**Comment by whitneywhtsang** [comment]
@pbchekin Can we use CI to do the measurements?

**Comment by pbchekin** [comment]
> @pbchekin Can we use CI to do the measurements?

Sure I will do.

**Comment by pbchekin** [comment]
Workflow runs (analysis to follow):

### baseline (rolling stable)

E2E performance: https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/10743997902

### open-linux-driver-verify-dev_igc-86345 (recompilation disabled)

Build and test: https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/10745120290
E2E performance: https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/10745633023

### open-linux-driver-ci-dev_igc-17627 (reference)

Build and test: https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/10745158801
E2E performance: https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/10745654834

**Comment by whitneywhtsang** [comment]
Can you please also measure key benchmarks (GEMM, flash attention, ...)?

**Comment by pbchekin** [comment]
# dev / ref

```
Overview of relative difference in speedup.
Relative difference 0.0 means both perform identically, relative difference > 0.0 means dev performs better, relative difference < 0.0 means ref performs better
count    500.000000
mean       0.007546
std        0.094587
min       -0.372935
25%       -0.008025
50%        0.003872
75%        0.017736
max        0.650195
Name: relative difference, dtype: float64
Mean speedup for denominator: 1.8044538440000002



10 fastest configurations (ref faster than dev, showing relative difference in speedup)
   dev        suite                            name       mode  datatype  speedup dev  speedup ref  relative difference
0  xpu  huggingface  MobileBertForQuestionAnswering  inference  amp_fp16     1.367795     2.181265            -0.372935
1  xpu  huggingface  MobileBertForQuestionAnswering  inference  amp_bf16     1.368590     2.138031            -0.359883
2  xpu   torchbench                    mobilenet_v2  inference  amp_fp16     1.327093     1.876985            -0.292966
3  xpu   torchbench                    mobilenet_v2  inference  amp_bf16     1.336445     1.883029            -0.290268
4  xpu  huggingface      T5ForConditionalGeneration   training  amp_bf16     0.593882     0.825136            -0.280262
5  xpu  huggingface      T5ForConditionalGeneration   training  amp_fp16     0.591901     0.821459            -0.279452
6  xpu  huggingface           MobileBertForMaskedLM  inference  amp_fp16     1.377309     1.897339            -0.274084
7  xpu  huggingface           MobileBertForMaskedLM  inference  amp_bf16     1.453354     1.973972            -0.263741
8  xpu   torchbench                     densenet121  inference  amp_bf16     1.842673     2.457808            -0.250278
9  xpu   torchbench                        resnet50  inference  amp_bf16     2.148037     2.783868            -0.228398



10 slowest configurations (ref slower than dev, showing relative difference in speedup)
   dev        suite                            name       mode  datatype  speedup dev  speedup ref  relative difference
0  xpu  huggingface  MobileBertForQuestionAnswering  inference  amp_fp16     2.144196     1.299359             0.650195
1  xpu  huggingface  MobileBertForQuestionAnswering  inference  amp_bf16     2.003114     1.342529             0.492045
2  xpu  huggingface           MobileBertForMaskedLM  inference  amp_fp16     1.925363     1.332076             0.445385
3  xpu   torchbench                     densenet121  inference  amp_bf16     2.547911     1.777027             0.433805
4  xpu   torchbench                    mobilenet_v2  inference  amp_fp16     1.876240     1.342696             0.397368
5  xpu   torchbench                    mobilenet_v2  inference  amp_bf16     1.873746     1.350724             0.387216
6  xpu   torchbench                     densenet121  inference  amp_fp16     2.536049     1.898708             0.335671
7  xpu   torchbench                        resnet50  inference  amp_bf16     2.801538     2.099387             0.334455
8  xpu   torchbench                        resnet50  inference  amp_fp16     2.812128     2.112672             0.331076
9  xpu  huggingface           MobileBertForMaskedLM  inference  amp_bf16     1.910963     1.439972             0.327083
```

**Comment by pbchekin** [comment]
# dev / ref

![image](https://github.com/user-attachments/assets/14d3d8b0-d14e-41b4-b5ff-e74f80368708)

**Comment by pbchekin** [comment]
# E2E performance base / dev

```
Overview of relative difference in speedup.
Relative difference 0.0 means both perform identically, relative difference > 0.0 means base performs better, relative difference < 0.0 means dev performs better
count    500.000000
mean      -0.006286
std        0.094013
min       -0.510157
25%       -0.024343
50%       -0.005784
75%        0.007461
max        0.626236
Name: relative difference, dtype: float64
Mean speedup for denominator: 1.810498386



10 fastest configurations (dev faster than base, showing relative difference in speedup)
   dev        suite                            name       mode  datatype  speedup base  speedup dev  relative difference
0  xpu   torchbench                        resnet50  inference  amp_bf16      1.372315     2.801538            -0.510157
1  xpu  huggingface  MobileBertForQuestionAnswering  inference  amp_fp16      1.254376     2.144196            -0.414990
2  xpu   torchbench                        resnet50  inference  amp_bf16      1.372315     2.148037            -0.361131
3  xpu  huggingface  MobileBertForQuestionAnswering  inference  amp_bf16      1.322812     2.003114            -0.339622
4  xpu   torchbench                    mobilenet_v2  inference  amp_bf16      1.327030     1.873746            -0.291777
5  xpu   torchbench                    mobilenet_v2  inference  amp_fp16      1.343251     1.876240            -0.284073
6  xpu   torchbench                     densenet121  inference  amp_bf16      1.824872     2.547911            -0.283777
7  xpu  huggingface           MobileBertForMaskedLM  inference  amp_fp16      1.429109     1.925363            -0.257746
8  xpu   torchbench                     densenet121  inference   float16      1.827480     2.458225            -0.256586
9  xpu   torchbench                     densenet121  inference  amp_fp16      1.900137     2.536049            -0.250749



10 slowest configurations (dev slower than base, showing relative difference in speedup)
   dev        suite                            name       mode  datatype  speedup base  speedup dev  relative difference
0  xpu  huggingface  MobileBertForQuestionAnswering  inference  amp_bf16      2.225651     1.368590             0.626236
1  xpu  huggingface  MobileBertForQuestionAnswering  inference  amp_fp16      2.015269     1.367795             0.473371
2  xpu   torchbench                    mobilenet_v2  inference  amp_fp16      1.890776     1.327093             0.424750
3  xpu   torchbench                    mobilenet_v2  inference  amp_bf16      1.896422     1.336445             0.419005
4  xpu   torchbench                     densenet121  inference  amp_bf16      2.494069     1.842673             0.353506
5  xpu   torchbench                        resnet50  inference  amp_bf16      2.856483     2.148037             0.329811
6  xpu  huggingface           MobileBertForMaskedLM  inference  amp_fp16      1.805213     1.377309             0.310681
7  xpu  timm_models                       fbnetv3_b  inference  bfloat16      2.492595     1.961467             0.270781
8  xpu  timm_models                       fbnetv3_b  inference  amp_bf16      2.323545     1.840114             0.262718
9  xpu   torchbench                        resnet50  inference  amp_fp16      2.759442     2.190877             0.259515
```

**Comment by pbchekin** [comment]
# E2E performance base / dev

![image](https://github.com/user-attachments/assets/e64f7613-ac4f-4cdc-94b2-2b6c42018f92)

**Comment by pbchekin** [comment]
# E2E performance base / ref

```
Overview of relative difference in speedup.
Relative difference 0.0 means both perform identically, relative difference > 0.0 means base performs better, relative difference < 0.0 means ref performs better
count    500.000000
mean      -0.002681
std        0.098260
min       -0.507047
25%       -0.018913
50%       -0.002567
75%        0.010441
max        0.657805
Name: relative difference, dtype: float64
Mean speedup for denominator: 1.8044538440000002



10 fastest configurations (ref faster than base, showing relative difference in speedup)
   dev        suite                            name       mode  datatype  speedup base  speedup ref  relative difference
0  xpu   torchbench                        resnet50  inference  amp_bf16      1.372315     2.783868            -0.507047
1  xpu  huggingface  MobileBertForQuestionAnswering  inference  amp_fp16      1.254376     2.181265            -0.424932
2  xpu  huggingface  MobileBertForQuestionAnswering  inference  amp_bf16      1.322812     2.138031            -0.381294
3  xpu   torchbench                        resnet50  inference  amp_bf16      1.372315     2.099387            -0.346326
4  xpu   torchbench                    mobilenet_v2  inference  amp_bf16      1.327030     1.883029            -0.295268
5  xpu   torchbench                    mobilenet_v2  inference  amp_fp16      1.343251     1.876985            -0.284357
6  xpu  huggingface      T5ForConditionalGeneration   training  amp_fp16      0.589581     0.821459            -0.282276
7  xpu  huggingface      T5ForConditionalGeneration   training  amp_bf16      0.597762     0.825136            -0.275559
8  xpu  huggingface           MobileBertForMaskedLM  inference  amp_bf16      1.451196     1.973972            -0.264835
9  xpu   torchbench                     densenet121  inference  amp_bf16      1.824872     2.457808            -0.257521



10 slowest configurations (ref slower than base, showing relative difference in speedup)
   dev        suite                            name       mode  datatype  speedup base  speedup ref  relative difference
0  xpu  huggingface  MobileBertForQuestionAnswering  inference  amp_bf16      2.225651     1.342529             0.657805
1  xpu  huggingface  MobileBertForQuestionAnswering  inference  amp_fp16      2.015269     1.299359             0.550972
2  xpu   torchbench                    mobilenet_v2  inference  amp_fp16      1.890776     1.342696             0.408194
3  xpu   torchbench                    mobilenet_v2  inference  amp_bf16      1.896422     1.350724             0.404004
4  xpu   torchbench                     densenet121  inference  amp_bf16      2.494069     1.777027             0.403507
5  xpu   torchbench                        resnet50  inference  amp_bf16      2.856483     2.099387             0.360627
6  xpu  huggingface           MobileBertForMaskedLM  inference  amp_fp16      1.805213     1.332076             0.355188
7  xpu   torchbench                        resnet50  inference  amp_fp16      2.759442     2.112672             0.306138
8  xpu  timm_models                       fbnetv3_b  inference  bfloat16      2.492595     1.939409             0.285234
9  xpu  timm_models                       fbnetv3_b  inference  amp_fp16      2.350860     1.848050             0.272076
```

**Comment by pbchekin** [comment]
# E2E performance base / ref
![image](https://github.com/user-attachments/assets/8e9b2bb0-51e5-4cf8-a5a3-6690a53c04c0)

**Comment by pbchekin** [comment]
# E2E performance base / dev

```
Overview of relative difference in speedup.
Relative difference 0.0 means both perform identically, relative difference > 0.0 means base performs better, relative difference < 0.0 means dev performs better
count    100.000000
mean      -0.010974
std        0.048393
min       -0.289045
25%       -0.018474
50%       -0.006533
75%        0.001199
max        0.272643
Name: relative difference, dtype: float64
Mean speedup for denominator: 1.5422985599999999



10 fastest configurations (dev faster than base, showing relative difference in speedup)
   dev        suite                        name      mode  datatype  speedup base  speedup dev  relative difference
0  xpu  huggingface  T5ForConditionalGeneration  training  amp_fp16      0.587848     0.826843            -0.289045
1  xpu   torchbench                       hf_T5  training  bfloat16      0.882893     1.003101            -0.119836
2  xpu   torchbench                    resnet50  training  amp_fp16      1.272964     1.417965            -0.102260
3  xpu  huggingface            XLNetLMHeadModel  training   float16      0.588321     0.642266            -0.083992
4  xpu   torchbench                mobilenet_v2  training  amp_bf16      1.888669     2.060620            -0.083446
5  xpu  timm_models               gmixer_24_224  training  amp_bf16      1.841845     1.997225            -0.077798
6  xpu   torchbench                mobilenet_v2  training  amp_fp16      1.944281     2.066511            -0.059148
7  xpu  huggingface            XLNetLMHeadModel  training  amp_bf16      0.789197     0.838756            -0.059086
8  xpu  huggingface            XLNetLMHeadModel  training  amp_fp16      0.789943     0.835214            -0.054203
9  xpu   torchbench                    resnet50  training  amp_bf16      1.380853     1.444161            -0.043837



10 slowest configurations (dev slower than base, showing relative difference in speedup)
   dev        suite                            name      mode  datatype  speedup base  speedup dev  relative difference
0  xpu  huggingface      T5ForConditionalGeneration  training   float32      0.659318     0.518070             0.272643
1  xpu  timm_models           beit_base_patch16_224  training   float16      1.132139     1.057693             0.070385
2  xpu  huggingface      T5ForConditionalGeneration  training  bfloat16      1.311724     1.234715             0.062370
3  xpu  huggingface      BlenderbotSmallForCausalLM  training  bfloat16      1.486295     1.411188             0.053223
4  xpu  huggingface           MobileBertForMaskedLM  training  amp_bf16      1.390892     1.341350             0.036934
5  xpu  huggingface  MobileBertForQuestionAnswering  training  amp_fp16      1.189265     1.159404             0.025755
6  xpu  huggingface           DistilBertForMaskedLM  training  bfloat16      1.248956     1.226372             0.018415
7  xpu  huggingface           MobileBertForMaskedLM  training  bfloat16      1.153312     1.137348             0.014036
8  xpu  huggingface  MobileBertForQuestionAnswering  training  amp_bf16      1.193186     1.178026             0.012869
9  xpu  huggingface           DistilBertForMaskedLM  training  amp_bf16      1.611947     1.596824             0.009471
```

![image](https://github.com/user-attachments/assets/88503402-fba1-4684-8d05-2f5fab6506db)

**Comment by pbchekin** [comment]
# E2E performance base / ref

```
Overview of relative difference in speedup.
Relative difference 0.0 means both perform identically, relative difference > 0.0 means base performs better, relative difference < 0.0 means ref performs better
count    100.000000
mean      -0.006914
std        0.025102
min       -0.117750
25%       -0.013841
50%       -0.004318
75%        0.001368
max        0.067873
Name: relative difference, dtype: float64
Mean speedup for denominator: 1.5375864699999997



10 fastest configurations (ref faster than base, showing relative difference in speedup)
   dev        suite                        name      mode  datatype  speedup base  speedup ref  relative difference
0  xpu   torchbench                       hf_T5  training  bfloat16      0.882893     1.000729            -0.117750
1  xpu   torchbench                    resnet50  training  amp_fp16      1.272964     1.421435            -0.104451
2  xpu   torchbench                mobilenet_v2  training  amp_bf16      1.888669     2.066609            -0.086102
3  xpu   torchbench                    resnet50  training  amp_bf16      1.380853     1.455754            -0.051452
4  xpu   torchbench                mobilenet_v2  training  amp_fp16      1.944281     2.045372            -0.049424
5  xpu  timm_models               gmixer_24_224  training  amp_bf16      1.841845     1.935273            -0.048276
6  xpu   torchbench                 densenet121  training   float16      0.965969     1.003795            -0.037683
7  xpu   torchbench                    resnet50  training  bfloat16      1.502486     1.561202            -0.037609
8  xpu  timm_models               gmixer_24_224  training  bfloat16      1.820687     1.883077            -0.033132
9  xpu  huggingface  T5ForConditionalGeneration  training  amp_bf16      0.812091     0.836023            -0.028626



10 slowest configurations (ref slower than base, showing relative difference in speedup)
   dev        suite                            name      mode  datatype  speedup base  speedup ref  relative difference
0  xpu   torchbench                        resnet50  training   float16      1.459940     1.367147             0.067873
1  xpu  timm_models           beit_base_patch16_224  training   float16      1.132139     1.076094             0.052082
2  xpu  huggingface      T5ForConditionalGeneration  training  bfloat16      1.311724     1.253105             0.046779
3  xpu  huggingface           MobileBertForMaskedLM  training  amp_bf16      1.390892     1.330478             0.045408
4  xpu   torchbench                     densenet121  training   float32      0.970261     0.943828             0.028006
5  xpu  huggingface           MobileBertForMaskedLM  training   float16      1.155468     1.127148             0.025125
6  xpu  huggingface                XLNetLMHeadModel  training   float16      0.588321     0.574104             0.024764
7  xpu  huggingface  MobileBertForQuestionAnswering  training  amp_bf16      1.193186     1.167982             0.021579
8  xpu  huggingface                XLNetLMHeadModel  training  amp_fp16      0.789943     0.775598             0.018495
9  xpu  huggingface           DistilBertForMaskedLM  training   float16      1.233545     1.213450             0.016560
```

![image](https://github.com/user-attachments/assets/2396353d-4293-4fe1-813a-6806f09a2b09)

**Comment by pbchekin** [comment]
# E2E performance dev/ ref

```
Overview of relative difference in speedup.
Relative difference 0.0 means both perform identically, relative difference > 0.0 means dev performs better, relative difference < 0.0 means ref performs better
count    100.000000
mean       0.006058
std        0.048371
min       -0.210554
25%       -0.005678
50%        0.000177
75%        0.008083
max        0.378111
Name: relative difference, dtype: float64
Mean speedup for denominator: 1.5375864699999997



10 fastest configurations (ref faster than dev, showing relative difference in speedup)
   dev        suite                            name      mode  datatype  speedup dev  speedup ref  relative difference
0  xpu  huggingface      T5ForConditionalGeneration  training   float32     0.518070     0.656245            -0.210554
1  xpu  huggingface      BlenderbotSmallForCausalLM  training  bfloat16     1.411188     1.466383            -0.037640
2  xpu  timm_models                   gmixer_24_224  training  bfloat16     1.839809     1.883077            -0.022977
3  xpu   torchbench                    mobilenet_v2  training  bfloat16     2.180193     2.224165            -0.019770
4  xpu  huggingface  MobileBertForQuestionAnswering  training  amp_fp16     1.159404     1.182565            -0.019585
5  xpu   torchbench                     densenet121  training   float16     0.984650     1.003795            -0.019073
6  xpu   torchbench                    mobilenet_v2  training   float16     2.127596     2.166300            -0.017866
7  xpu  huggingface              ElectraForCausalLM  training  amp_fp16     1.809878     1.842128            -0.017507
8  xpu  timm_models           beit_base_patch16_224  training   float16     1.057693     1.076094            -0.017100
9  xpu  timm_models                       fbnetv3_b  training  bfloat16     2.035677     2.069183            -0.016193



10 slowest configurations (ref slower than dev, showing relative difference in speedup)
   dev        suite                        name      mode  datatype  speedup dev  speedup ref  relative difference
0  xpu  huggingface  T5ForConditionalGeneration  training  amp_fp16     0.826843     0.599983             0.378111
1  xpu  huggingface            XLNetLMHeadModel  training   float16     0.642266     0.574104             0.118728
2  xpu   torchbench                    resnet50  training   float16     1.480997     1.367147             0.083276
3  xpu  huggingface            XLNetLMHeadModel  training  amp_fp16     0.835214     0.775598             0.076865
4  xpu  huggingface            XLNetLMHeadModel  training  amp_bf16     0.838756     0.783966             0.069888
5  xpu   torchbench                 densenet121  training   float32     0.980172     0.943828             0.038507
6  xpu  huggingface  T5ForConditionalGeneration  training   float16     1.363572     1.314613             0.037242
7  xpu   torchbench                       hf_T5  training   float16     1.488868     1.439170             0.034532
8  xpu  timm_models               gmixer_24_224  training  amp_bf16     1.997225     1.935273             0.032012
9  xpu  huggingface  BlenderbotSmallForCausalLM  training  amp_fp16     1.717258     1.665488             0.031084
```

![image](https://github.com/user-attachments/assets/ffc0c21a-a9bd-4700-9c69-94577151524e)

**Comment by pbchekin** [comment]
# Softmax performance dev / ref

```
Overview of relative difference in tri2xe.
Relative difference 0.0 means both perform identically, relative difference > 0.0 means dev performs better, relative difference < 0.0 means ref performs better
count    7.000000
mean    -0.003552
std      0.120562
min     -0.158170
25%     -0.076779
50%      0.011529
75%      0.033806
max      0.207722
Name: relative difference, dtype: float64
Mean tri2xe for denominator: 1.1976769249953823



10 best configurations (ref better than dev, showing relative difference in tri2xe)
                           Case  Tri2Xe-dev  Tri2Xe-ref  relative difference
0   softmax-performance: 2048.0    1.038080    1.233123            -0.158170
1   softmax-performance: 4096.0    1.357726    1.553066            -0.125777
2   softmax-performance: 8192.0    1.042843    1.072641            -0.027780
3  softmax-performance: 16384.0    1.025174    1.013489             0.011529
4  softmax-performance: 32768.0    0.993479    0.968341             0.025960
5   softmax-performance: 1024.0    1.324441    1.271482             0.041652
6    softmax-performance: 256.0    1.535734    1.271596             0.207722



10 worst configurations (ref worse than dev, showing relative difference in tri2xe)
                           Case  Tri2Xe-dev  Tri2Xe-ref  relative difference
0    softmax-performance: 256.0    1.535734    1.271596             0.207722
1   softmax-performance: 1024.0    1.324441    1.271482             0.041652
2  softmax-performance: 32768.0    0.993479    0.968341             0.025960
3  softmax-performance: 16384.0    1.025174    1.013489             0.011529
4   softmax-performance: 8192.0    1.042843    1.072641            -0.027780
5   softmax-performance: 4096.0    1.357726    1.553066            -0.125777
6   softmax-performance: 2048.0    1.038080    1.233123            -0.158170
```

**Comment by pbchekin** [comment]
# Matmul performance dev/ ref

## dev
```
B	M	K	N	Triton-GB/s
1	1024	1024	1024	176.87708
1	2048	2048	2048	233.410696
1	4096	4096	4096	121.715146
1	8192	8192	8192	63.704497
1	1	5120	13824	535.332492
1	4	4096	12288	570.896747
1	512	8192	8192	283.311581
1	512	8192	32768	285.819527
1	512	32768	8192	257.983987
1	1024	16384	8192	160.045179
1	1024	28672	8192	139.603161
1	3072	4096	3072	116.014309
1	4096	16384	8192	62.639235
1	8192	16384	1024	155.464071
1	8192	16384	4096	60.756195
1	16384	1024	8192	268.307897
1	16384	4096	8192	86.551876
1	16384	8192	1024	163.11936
1	16384	8192	4096	68.958797
4	32768	128	4096	1045.795907
4	32768	4096	128	602.934521
32	4096	4096	128	609.390451
4096	8	128	16384	853.923865
4096	8	16384	128	903.353991
```

## ref
```
B	M	K	N	Triton-GB/s
1	1024	1024	1024	181.296903
1	2048	2048	2048	233.696028
1	4096	4096	4096	121.726186
1	8192	8192	8192	63.729753
1	1	5120	13824	537.853533
1	4	4096	12288	569.814285
1	512	8192	8192	284.222209
1	512	8192	32768	287.405129
1	512	32768	8192	259.099581
1	1024	16384	8192	159.50047
1	1024	28672	8192	139.715588
1	3072	4096	3072	116.047945
1	4096	16384	8192	62.769598
1	8192	16384	1024	155.245404
1	8192	16384	4096	60.672158
1	16384	1024	8192	268.165742
1	16384	4096	8192	86.923049
1	16384	8192	1024	163.376545
1	16384	8192	4096	69.248806
4	32768	128	4096	1055.076528
4	32768	4096	128	608.005697
32	4096	4096	128	612.728851
4096	8	128	16384	853.92926
4096	8	16384	128	906.051701
```

**Comment by pbchekin** [comment]
TLDR: from the results above i do not see significant performance difference between dev and ref versions. There are several spikes (in favor of both dev and ref) that can can be explained by the existing test environment. In my opinion, variations with average performance less than 10% can be ignored.

* In average, dev shows 6% better performance than ref with E2E performance benchmarks.
* In average, ref shows 4% better performance than dev with softmax.
* In average, ref and def show the same performance (<1%) with matmul.

**Comment by pbchekin** [comment]
@esukhov let me know if the results are good.

**Comment by esukhov** [comment]
Thanks a lot, that's a great news actually. Could you please give me the dumps for the workloads with the biggest difference  between versions? Both dev & ref? 
I will go through them, maybe we can combine the best of both worlds.

**Comment by pbchekin** [comment]
> Could you please give me the dumps for the workloads with the biggest difference between versions?

What dumps do you mean?

**Comment by whitneywhtsang** [comment]
> > Could you please give me the dumps for the workloads with the biggest difference between versions?
> 
> What dumps do you mean?

I think he meant shader dump, which can be obtained with env vars e.g. `export IGC_ShaderDumpEnable=1 IGC_DumpToCustomDir=./mm_dumps`.

**Comment by pbchekin** [comment]
# ref faster than dev: huggingface, training, float32, T5ForConditionalGeneration

* [ref-huggingface-training-float32-T5ForConditionalGeneration.zip](https://modin-datasets.intel.com/triton-xpu/ref-huggingface-training-float32-T5ForConditionalGeneration.zip)
* [dev-huggingface-training-float32-T5ForConditionalGeneration.zip](https://modin-datasets.intel.com/triton-xpu/dev-huggingface-training-float32-T5ForConditionalGeneration.zip)

# ref slower than dev: huggingface, training, amp_fp16, T5ForConditionalGeneration

* [ref-huggingface-training-amp_fp16-T5ForConditionalGeneration.zip](https://modin-datasets.intel.com/triton-xpu/ref-huggingface-training-amp_fp16-T5ForConditionalGeneration.zip)
* [dev-huggingface-training-amp_fp16-T5ForConditionalGeneration.zip](https://modin-datasets.intel.com/triton-xpu/dev-huggingface-training-amp_fp16-T5ForConditionalGeneration.zip)

**Comment by pbchekin** [comment]
@esukhov please find IGC dumps above for the cases with the biggest difference.
